File: rb/features.rb
# IpAddressLookupTwo SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module IpAddressLookupTwoFeatures
  def self.make_feature(name)
    case name
    when "base"
      IpAddressLookupTwoBaseFeature.new
    when "test"
      IpAddressLookupTwoTestFeature.new
    else
      IpAddressLookupTwoBaseFeature.new
    end
  end
end

File: py/ipaddresslookuptwo_sdk/features.py
# IpAddressLookupTwo SDK feature factory

from ipaddresslookuptwo_sdk.feature.base_feature import IpAddressLookupTwoBaseFeature
from ipaddresslookuptwo_sdk.feature.test_feature import IpAddressLookupTwoTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IpAddressLookupTwoBaseFeature(),
        "test": lambda: IpAddressLookupTwoTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
